test (#4)
annie-samsung/tizen-docs: docs/application/tizen-studio-0/setup/test.md

> # Grid layout(Tizen 6.0 or late)

Reviewer: Add a space between word "layout" and the first parenthesis.
Also, replace the word "late" with "later".

> # Grid layout(Tizen 6.0 or late)

Reviewer: Change the first letter to upper case in the word "layout".

> > **Note**

Reviewer: Change the note format to the latest as below:

[!Note]

Reviewer 2: @sagarkedari  Correct form is: 

> [!NOTE]
> Text 

This means you need to ask to remove > in middle. 
And, The note is all caps

>

> > If you are using GridLayout on Tizen 5.5, Please refer to [Grid Layout(Tizen 5.5)](#gridlayout-tizen5.5) instead.

Reviewer: Change first letter to lower case in the word "Please".

> `GridLayout` is a grid box for the two dimentional layout. It constraints the x and y position, width and height of the child actors.

Reviewer: Spelling error : change "dimentional" to "dimensional".

Also, please change the second sentence as below:
It constraints the x and y positions, width, and height of the child actors.

> Positioning children in a grid form, the size of each column and row is set to the largest size of cells belonging to their axis.

Reviewer: Can we change the condition and sentence as below?

To position children in a grid form, set the size of each column and row to the largest size of cells belonging to their axis.

Reviewer 2: `GridLayout` is a grid box for the two-dimensional layout. It constraints the x and y position, width and height of the child actors.  `GridLayout` positions the children in a grid form, where the size of each column and row is set to the largest size of the cells belonging to their axis.

> If the `Column` or the `row` Property is not specified on the child, it is automtically assigned left to right or top to bottom depends on the `GridOrientation` Property.

Reviewer: Change first letter to upper case in the word "row". Also, change first letter to lower case in the word "Property" at both the places in the sentence.

Spelling error : change "automtically" as "automatically" 

Can we phrase the sentence as below?
If the `Column` or `Row` property is not specified on the child, `GridLayout` automatically assigns left to right or top to bottom, depending on the `GridOrientation` property.

> There are two types of properties. One is the properties for the grid layout and Another one is the properties that affect the specified child:

Reviewer: Replace "Another one" with "the other" in the sentence.

Also change grid layout to `GridLayout`

Reviewer 2: Here the ist sentence is incomplete, we can write: 

There are two types of properties that affect the `GridLayout`. One is the properties for the grid layout and the other is the properties that affect the specified child.

> The following table describes the properties of `GridLayout`.

Reviewer 2: Missed colon at end instead of .

| Property               | Type            | Description |
| -----------------------| --------------- | ------------ |

> | `Columns`              | int             | Gets or sets the number of colomns in the grid. |

Reviewer: Spell Error : Change "colomns" to "columns"

| `Rows`                 | int             | Gets or sets the number of rows in the grid. |
| `GridOrientation`      | Orientation     | Gets or sets the orientation in the grid. |

> | `ColumnSpacing         | `float`         | Gets or sets the distance between columns |

Reviewer 2: Code style missing

| `RowSpacing`           | float           | Gets or sets the distance between rows. |

> <a name="columns-and-rows"></a>

Reviewer: Not sure why this line has been added?

> ## Columns and rows

Reviewer 2: @sagarkedari We follow sentence case, so you can write like this: 

Global: Use sentence case for all headings. For reference : https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md

The `Columns` and the `Rows` properties specify the maximum number of cells on a horizontal or vertical axis.

> When the `Column` or the `Row` of child is automatically set without specified value, the value cannot be exceeded the number of main-axis cell which is the `Columns` value on horizontal orientation or the `Rows` value on vertical orientation. But, it can be assigned a value that exceeds the number of cross-axis cell.

Reviewer: Update sentence as below:

When the `Column` or the `Row` of child is automatically set without specified value, the value cannot exceed the number of main-axis cell which is (rest is fine in the sentence)

Reviewer 2: But doesn't go well in the sentence , so consider changing it. 

However, ....it can be assigned a value that exceeds the number of cross-axis cell.

> The following figures shows how to set the `Colomns` and the `Rows` properties. Since the `GridOrientation` is set to horizontal, the row index of last item can be `2` that exceeds the `Rows` property.

Reviewer: Change "figures" to figure, since there is only one figure included.

Spell error : Change `Colomns` to `Columns`

Reviewer 2: @sagarkedari 
```suggestion
The following figure shows how to set the `Columns` and the `Rows` properties. Since the `GridOrientation` is set to horizontal, the row index of the last item can be `2` that exceeds the `Rows` property:
```

Point to be noted: 

Whenever the sentence is followed up by an image, table or code segment we use a colon instead of a period.

> ![Column](./media/columnLayout.png)

Reviewer: update file name inside the square bracket to [columnLayout]

```csharp
View layoutView = new View();
var gridLayout = new GridLayout();
gridLayout.GridOrientation = GridLayout.Orientation.Horizontal;
gridLayout.Columns = 2
gridLayout.Rows = 1
layoutView.Layout = gridLayout;
```

> <a name="grid-orientation"></a>

Reviewer: Not sure why this line has been added?

## Grid Orientation

> The `GridOrientation` only works when a child is added without the specified `Row` or `Column` property.

Reviewer: Suggestion: Can we change the sentence as below: 

The `GridOrientation` only works when a child is added without specifying the `Row` or `Column` property.

OR

The `GridOrientation` only works when a child is added without the specific `Row` or `Column` property.

Reviewer 2: @sagarkedari  
If you are in doubt, it's better to ask the developer to elaborate things. 

After you understand, then you can give your suggestion. And whenever you give a suggestion/rephrase ensure you don't change the meaning.

>  - On horizontal orientation, the children will autmatically occupy empty cells from left to right.

Reviewer: Spell error : 

Change "autmatically" as "automatically"

Also, please add article "the" before word "empty".

Reviewer 2: @sagarkedari We dont use **will** in our docs, where ever you find future tense, change it to present or present perfect tense.

>  - On vertical orientation, the children will automatically occupy empty cells from top to bottom.

Reviewer: Please add article "the" before word "empty".

Reviewer 2: @sagarkedari We dont use **will** in our docs, where ever you find future tense, change it to present or present perfect tense.

Reviewer 2: On horizontal orientation, the children automatically occupy the empty cells from left to right.

The following figure shows how the `GridOrientation` property works:

> ![gridOrientation](./media/gridOrientatio.png)

Reviewer: Add the last letter in the file name.

./media/gridOrientation.png

## Column spacing and row spacing

> The `ColumnSpacing` and `rowSpacing` properties are the distance between columns or rows. In order to set same sapce between the children, you can use `ColumnSpacing` and `rowSpacing` properties instead of setting the same margin for each child.

Reviewer: Change `rowSpacing` to `RowSpacing`

Change "distance" to "distances"

Spell error: Change "sapce" to "space". Also add article "the" before word same

Reviewer 2: ```suggestion
The `ColumnSpacing` and `rowSpacing` properties are the distance between columns or rows. In order to set the same space between the children, you can use `ColumnSpacing` and `rowSpacing` properties instead of setting the same margin for each child.
```

What made you change the property name, did you refer to something ? Or you thought r must be capital. Property names must be written as is in the code, if you don't know about the property name base page ask the developer to guide you to reach the API/ class names.

Precisely `ColumnSpacing` and `rowSpacing`  are property names. These must be written as per their usage in code or API files.

In line 56 r is small, and in 58 r is capital. So ask developer what is correct way.

The following figure shows how the `ColumnSpacing` and `RowSpacing` properties work:

![gridSpacing](./media/gridSpacing.png)

> The following table describes the properties that affect the specified child.

Reviewer: Change "specified" to "specific"

| Property               | Type            | Description |
| -----------------------| --------------- | ----------- |
| `Column`               | int             | Gets or sets the column index of the child. |
| `Row`                  | int             | Gets or sets the row index of the child. |
| `ColumnSpan`           | int             | Gets or sets the number of columns that the child occupy. |
| `RowSpan`              | int             | Gets or sets the number of rows that the child occupy. |
| `HorizontalStretch`    | StretchFlags    | Gets or sets the value how child will be resized within horizontal space of the cell where child occupy. |
| `VerticalStretch`      | StretchFlags    | Gets or sets the value how child is resized within vertical space of the cell where child occupy. |
| `HorizontalAlignment`  | Alignment       | Gets or sets the horizontal alignment of the child within the cell. |
| `VerticalAlignment`    | Alignment       | Gets or sets the vertical alignment of the child within the cell. |

## Column and row

> The `Column` and `row` properties specify the cell index of the child. If these properties aren't assigned, they are automaticaly set dependig on [`GridOrientation`](#grid-orientation), [`Columns`](#columns-and-rows) and [`Rows`](#columns-and-rows).

Reviewer: Plse change `row` to `Row`

Also, please write the word completely : "aren't" as "are not"

Spell error : change "dependig" as "depending"

Reviewer 2: @sagarkedari
-  Missed automatically spell. 
- Serial comma missing before and. 

You must not change property name without referencing its code or API usage, `row` is correct until you check n verify `Row` is used. No guesswork.

Reviewer 2: Check here:https://samsung.github.io/TizenFX/API9/api/Tizen.NUI.GridLayout.html#properties

Please use these parameter names from the API

Reviewer 2: As both Columns and Rows are going to the same location, #columns-and-rows,
how about changing it to below?
[`Columns` and `Rows`](#columns-and-rows)

## Column span and row span

> The `ColumnSpan` and `rowSpan` properties the number of cell that the child occupy. The sum of cell index and cell span should be less than maximun number of cell specified by `Columns` or `Rows` Properties.

Reviewer: Plse change `rowSpan` to `RowSpan`

Plse update first sentence as below: 
The `ColumnSpan` and `RowSpan` properties specify the number of cell that the child occupy.

Also, add word "the" before word "maximum".

Reviewer 2: ```suggestion
The `ColumnSpan` and `rowSpan` properties the number of cells that the child occupies. The sum of cell index and cell span should be less than the maximum number of the cell specified by `Columns` or `Rows` Properties.
```
Missed spell maximum.

The following figure shows how to position child on the grid using `Column` and `ColumnSpan` properties:

![gridColumn](./media/gridColumn.png)

## Horizontal stretch and vertical stretch

> The `HorizontalStretch` and `VerticalStretch` properties define how child is resized within space of the cell. the value can be `Stretchflags` enum:

Reviewer: The last sentence looks incomplete. Can you please check and update?

Also, there is no reference to image in the topic. Please check and add the same.

Reviewer 2: ```suggestion
The `HorizontalStretch` and `VerticalStretch` properties define how the child is resized within the space of the cell. The values can be specified as `Stretchflags` enum:

![gridStretch](./media/gridStretch.png)

| Value           | Description |
| --------------- | ----------- |
| `None`          | Fits to the measured size. |
| `Fill`          | Resizes to completely fill the space. |
| `Expand`        | Expands to share available space. |

> | `ExpandAndFill` | Expands to share available space and fill the space.

Reviewer 2: @sagarkedari 
Missed | of the table. 
```suggestion
| `ExpandAndFill` | Expands to share available space and fill the space. |

## Horizontal alignment and vertical alignment

> The `HorizontalAlignment` and `VerticalAlignment` properties specify the alignment for the child within its cells:

Reviewer: Replace colon (:) with period (.) at the end of the sentence.

Reviewer 2: @sagarkedari Here we use : . Whenever there is code, image or table which immediately follows up a sentence we use :.

> ![gridAlignment](./media/gridAlignment.png)

Reviewer: There is no reference to this image in the topic. Please check for missing content and include the same.

> <a name="gridlayout-tizen5.5"></a>

Reviewer: Not sure why this has been added?

> # Grid Layout(Tizen 55)

Reviewer: Add space between word "Layout" and the first parenthesis.
Also, change Tizen 55 as Tizen 5.5

`GridLayout` is a grid box for the two dimensional layout. It constraints the x and y position, width, and height of the child actors.

> Positioning children in a grid form, the cells are of uniform size based on the first child added to the parent View.

Reviewer: This sentence doesn't look phrased properly, not sure what we are trying to convey here? could you please check and update?

You can set the number of **columns**, however the rows automatically increase to hold the children. After the available space is used, the remaining rows become invisible by default.

> 

Reviewer: Can we add a reference to below table for consistency?

> | Property               | Type            | Descrption |

Reviewer: Can we add a reference to this table for consistency?

> | Property               | Type            | Descrption |

Reviewer: Spell error : change "Descrption" as "Description"

| -----------------------| --------------- | ------------ |
| `Columns`              | int             | Gets or sets the number of columns in the grid. |

> ## Colomn

Reviewer 2: ## Column

> ![Column](./media/columnLayout.png)

Reviewer: Update file name inside the square bracket as [columnLayout]

```csharp
View layoutView = new View();
var gridLayout = new GridLayout();
gridLayout.Columns = 2;
layoutView.Layout = gridLayout;
```

> ## Related Information
> 
> * Dependencies
>   *  Tizen 5.5 and Higher

Reviewer 2: ```suggestion
## Related Information

- Dependencies
  -  Tizen 5.5 and Higher
 ```
We always use - instead of * or +
